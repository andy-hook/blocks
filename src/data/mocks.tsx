import { SocialMeta } from "@custom-types/model"
import { FluidObject } from "gatsby-image"
import { TransitionType } from "@custom-types/gatsby-plugin-transition-link"
import { TRANSITION_STATUS_ENTERING } from "@hooks/page-transition"

export const mockSocialIcons: SocialMeta = {
  email: {
    label: "Email",
    url: "email address",
    icon: "mail",
  },
  twitter: {
    label: "Twitter",
    url: "path/to/social/profile",
    icon: "twitter",
  },
  instagram: {
    label: "Instagram",
    url: "path/to/social/profile",
    icon: "instagram",
  },
  linkedin: {
    label: "Linkedin",
    url: "path/to/social/profile",
    icon: "linkedin",
  },
  dribbble: {
    label: "Dribbble",
    url: "path/to/social/profile",
    icon: "dribbble",
  },
  github: {
    label: "Github",
    url: "path/to/social/profile",
    icon: "github",
  },
}

export const mockFluidImageObject: FluidObject = {
  aspectRatio: 1.5,
  sizes: "(max-width: 1000px) 100vw, 1000px",
  src: "/static/4a7de89fcee5d7cd97c940895602fbe0/2f7e7/cover-image.jpg",
  srcSet:
    "/static/4a7de89fcee5d7cd97c940895602fbe0/4d406/cover-image.jpg 250w,↵/static/4a7de89fcee5d7cd97c940895602fbe0/32ee9/cover-image.jpg 500w,↵/static/4a7de89fcee5d7cd97c940895602fbe0/2f7e7/cover-image.jpg",
}

export const createStateMock = (animType?: TransitionType) => {
  return {
    current: {
      delay: 0,
      length: 0,
      state: {},
    },
    entry: {
      delay: 0,
      length: 0,
      state: {
        animType,
      },
    },
    exit: {
      delay: 0,
      length: 0,
      state: {
        animType,
      },
    },
    transitionStatus: TRANSITION_STATUS_ENTERING,
    mount: false,
  }
}
