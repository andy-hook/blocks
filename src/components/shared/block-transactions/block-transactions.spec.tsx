import React from "react"
import BlockTransactions from "./block-transactions"
import { renderWithTheme } from "@test-utils"
import { mockMultipleWeb3TransactionData } from "@data/mocks"

jest.mock("react-truncate-string")

describe("<BlockSingleTrx />", () => {
  it("renders correctly", () => {
    const tree = renderWithTheme(
      "dark",
      <BlockTransactions transactions={mockMultipleWeb3TransactionData} />
    )
    expect(tree).toMatchSnapshot()
  })
})
