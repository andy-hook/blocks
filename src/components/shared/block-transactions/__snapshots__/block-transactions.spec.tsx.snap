// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlockSingleTrx /> renders correctly 1`] = `
<table
  className="sc-bZQynM DcOju"
>
  <thead>
    <tr
      className="sc-bdVaJa bDWFJH"
    >
      <th
        className="sc-bwzfXH sc-htpNat KtFsv"
      >
        hello
      </th>
      <th
        className="sc-bwzfXH sc-bxivhb bUZGRr"
      >
        hello
      </th>
      <th
        className="sc-bwzfXH sc-ifAKCX cFlEyZ"
      >
        hello
      </th>
      <th
        className="sc-bwzfXH sc-EHOje jvzENE"
      >
        hello
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="sc-bdVaJa bDWFJH sc-gzVnrw daUmfx"
    >
      <td
        className="sc-bwzfXH sc-htpNat KtFsv"
      >
        7250676
      </td>
      <td
        className="sc-bwzfXH sc-bxivhb bUZGRr"
      >
        0x414a1b09ff2a1cafb2ddd13ed1cc229948ee349b7accd4ad3c3b250e46cddb8d
      </td>
      <td
        className="sc-bwzfXH sc-ifAKCX cFlEyZ"
      >
        <div>
          0xcf5f58c3E7d863c3c859C6Cbce03e015262df0e5
          0xF39d8b9B8C130ac08d4A32E50bb7bbD41C054EB9
        </div>
      </td>
      <td
        className="sc-bwzfXH sc-EHOje jvzENE"
      >
        1
      </td>
    </tr>
    <tr
      className="sc-bdVaJa bDWFJH sc-gzVnrw daUmfx"
    >
      <td
        className="sc-bwzfXH sc-htpNat KtFsv"
      >
        7250676
      </td>
      <td
        className="sc-bwzfXH sc-bxivhb bUZGRr"
      >
        0x414a1b09ff2a1cafb2ddd13ed1cc229948ee349b7accd4ad3c3b250e46cddb8d
      </td>
      <td
        className="sc-bwzfXH sc-ifAKCX cFlEyZ"
      >
        <div>
          0xcf5f58c3E7d863c3c859C6Cbce03e015262df0e5
          0xF39d8b9B8C130ac08d4A32E50bb7bbD41C054EB9
        </div>
      </td>
      <td
        className="sc-bwzfXH sc-EHOje jvzENE"
      >
        1
      </td>
    </tr>
  </tbody>
</table>
`;
